common/djangoapps/config_models/models.py

```diff
@@ -176,3 +176,46 @@ def key_values(cls, *key_fields, **kwargs):
         values = list(cls.objects.values_list(*key_fields, flat=flat).order_by().distinct())
         cache.set(cache_key, values, cls.cache_timeout)
         return values
+
+    def fields_equal(self, instance, fields_to_ignore=("id", "change_date", "changed_by")):
+        """
+        Compares this instance's fields to the supplied instance to test for equality.
+        This will ignore any fields in `fields_to_ignore`.
+
+        Args:
+            instance: the model instance to compare
+            fields_to_ignore: List of fields that should not be compared for equality. By default
+            includes `id`, `change_date`, and `changed_by`.
+
+        Returns: True if the checked fields are all equivalent, else False
+        """
+        for field in self._meta.get_fields():
+            if field.name not in fields_to_ignore and getattr(instance, field.name) != getattr(self, field.name):
+                return False
+
+        return True
+
+    @classmethod
+    def equal_to_existing_entry(cls, json, fields_to_ignore=("id", "change_date", "changed_by")):
+        """
+        Compares this instance to a model instance constructed from the supplied JSON for equality.
+        This will ignore any fields in `fields_to_ignore`.
+
+        Note that this method cannot handle fields with many-to-many associations, as those can only
```

Comment:
I went ahead and wrote the code (and a unit test) to ignore the many-to-many relationships. I'm not entirely sure that is the desired behavior (perhaps it would be better to error if there are many-to-many relationships?), but it can always be revisited if it turns out that such a use case does exist.